akka-http-core/src/main/scala/akka/http/impl/engine/ws/Masking.scala

```diff
@@ -51,12 +51,9 @@ private[http] object Masking {
       def onPush(part: FrameEvent, ctx: Context[FrameEvent]): SyncDirective =
         part match {
           case start @ FrameStart(header, data) ⇒
-            if (header.length == 0) ctx.push(part)
-            else {
```

Comment:
Btw. this was undetected before because the logic is only run for the client side of the autobahn test suite which wasn't run before. My own tests for the same direction made the same wrong assumption as the old code (because masking is somewhat useless if you mask zero bytes, it only adds 4 bytes of overhead to the frame, still it is more consistent to require it all the time) :)